src/core/qgsziputils.cpp

```diff
@@ -84,8 +85,19 @@ bool QgsZipUtils::unzip( const QString &zipFilename, const QString &dir, QString
         char *buf = new char[len];
         if ( zip_fread( file, buf, len ) != -1 )
         {
-          QFileInfo newFile( QDir( dir ), QString( stat.name ) );
-          std::ofstream( newFile.absoluteFilePath().toStdString() ).write( buf, len );
+          QString fileName( stat.name );
+          QFileInfo newFile( QDir( dir ), fileName );
+          bool isDirectory = fileName.endsWith( '/' );
+
+          if ( isDirectory )
+          {
+            if ( !QDir( dir ).mkdir( fileName ) )
+              QgsDebugMsg( QString( "Failed to create a subdirectory %1/%2" ).arg( dir ).arg( fileName ) );
+          }
+          else
+          {
+            std::ofstream( newFile.absoluteFilePath().toStdString() ).write( buf, len );
```

Comment:
Why not use `QFile::write`?

Reply:
I kept it as it was before, just moved to else branch.

Reply:
CC @pblottiere